src/nvim/mbyte.c

```diff
@@ -602,14 +602,14 @@ size_t mb_string2cells_len(const char_u *str, size_t size)
 /// @param[in]  p  String to convert.
 ///
 /// @return Unicode codepoint or byte value.
-int utf_ptr2char(const char_u *const p)
+int utf_ptr2char(const char *const p)
   FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
 {
-  if (p[0] < 0x80) {  // Be quick for ASCII.
+  if ((uint8_t)p[0] < 0x80) {  // Be quick for ASCII.
```

Comment:
I wonder if we want to do this or rather do something like change `p` to `p_in` in the arg name and then `uint8_t p = (uint8_t *)p_in` on the first row. Less risk for doing mistakes (clangd warnings might not catch everything, and it is nothing wrong with regarding a string as an array of unsigned chars when we want to).

Reply:
Yep, you're right. The tests are already failing and I think this should fix it. Thanks.

Also realized I didn't have [this warning](https://clang.llvm.org/extra/clang-tidy/checks/hicpp-signed-bitwise.html) enabled. This will help save some time in the future :)